packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -510,6 +525,8 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
             this.onTermDidClose.fire(this);
             this.onTermDidClose.dispose();
         }
+        this.onTermDidDispose.fire(undefined);
```

Comment:
are you sure that base class does not have an event or signal already?

Reply:
I checked `BaseWidget` (in theia) and `Widget` (in phorsphor), the closet event that I can find is `Widget.onCloseRequest()`. Is it the right one to use?

Reply:
ok, let's go with `onTermDidDispose` for now

Reply:
I thought all widgets have `onDispose` event already